tests/Cache/FileHandlerTest.php

```diff
@@ -145,6 +148,76 @@ public function testWriteWritesCache()
         $this->assertSame($cache->toJson(), $actualCacheJson);
     }
 
+    public function testWriteCacheToDirectory()
+    {
+        $dir = __DIR__.'/../Fixtures/cache-file-handler';
+
+        $handler = new FileHandler($dir);
+
+        $this->setExpectedExceptionRegExp(
+            'Symfony\Component\Filesystem\Exception\IOException',
+            sprintf('#^%s$#', preg_quote('Cannot write cache file "'.realpath($dir).'" as the location exists as directory.', '#')
+        ));
+
+        $handler->write(new Cache(new Signature(
+            PHP_VERSION,
+            '2.0',
+            array(
+                'foo',
+                'bar',
+            )
+        )));
+    }
+
+    public function testWriteCacheToNonWriteableFile()
+    {
+        $file = __DIR__.'/../Fixtures/cache-file-handler/cache-file';
```

Comment:
shouldn't this file be created/destroyed by test suite itself ?

Reply:
with proper preconfiguration and cleanup, we don't need to mark this test as skipped

Reply:
The problem is that the tests runs in one PHP process.
Creating a file within that process and than make it impossible to write to using permissions is (near) impossible on *ux systems. 
The alternative would be to setup the fixture file outside the PHP process. This should than be done for each CI and will fragile and require additional action when running the test locally.

Reply:
huh? you mean one cannot set permissions during execution? or I don't understand you

manual setup outside phpunit is not desired (except setup done like in `CiIntegrationTest`)

Reply:
I'm saying it more trouble than its worth to go through all edge cases based on OS deployments and user account used. This is a fixture file like any other fixture we've in the repo, I don't see why the test or CI env. the test is executed in has to create the file.